jujuclient/controllers.go

```diff
@@ -72,4 +72,10 @@ type Controllers struct {
 
 	// CurrentController is the name of the active controller.
 	CurrentController string `yaml:"current-controller,omitempty"`
+
+	// PreviousController is the name of the previous active controller.
+	PreviousController string `yaml:"previous-controller,omitempty"`
+
+	// HasControllerChangedOnPreviousSwitch represents whether the controller has changed during the previous switch.
+	HasControllerChangedOnPreviousSwitch bool `yaml:"has-controller-changed-on-previous-switch"`
```

Comment:
Is there really no way to improve this?

I can't think of any of the top of my head. But something else would be good. Even if it results in redundancy

Maybe `global-previous-model` with full `$controller:$model` format?

Reply:
I thought about this solution, but i don't like it, because it add "model" information in the controller list context. It doesn't feels right.

Having a boolean which state that the previous switch was extra or intra controller is still bounded to controllers level (whatever the actual content of a controller), so even if it's ugly, I think it's a better choice.